runtime/parser/errors.go

```diff
@@ -91,6 +91,60 @@ func (e *SyntaxError) Error() string {
 	return e.Message
 }
 
+// SyntaxErrorWithSuggestedFix
+
+type SyntaxErrorWithSuggestedFix struct {
+	Message      string
+	SuggestedFix string
+	Pos          ast.Position
+}
+
+var _ errors.HasSuggestedFixes[ast.TextEdit] = &SyntaxErrorWithSuggestedFix{}
+
+func NewSyntaxErrorWithSuggestedFix(pos ast.Position, message string, suggestedFix string) *SyntaxErrorWithSuggestedFix {
+	return &SyntaxErrorWithSuggestedFix{
+		Pos:          pos,
+		Message:      message,
+		SuggestedFix: suggestedFix,
+	}
+}
+
+var _ ParseError = &SyntaxErrorWithSuggestedFix{}
+var _ errors.UserError = &SyntaxErrorWithSuggestedFix{}
+
+func (*SyntaxErrorWithSuggestedFix) isParseError() {}
+
+func (*SyntaxErrorWithSuggestedFix) IsUserError() {}
+
+func (e *SyntaxErrorWithSuggestedFix) StartPosition() ast.Position {
+	return e.Pos
+}
+
+func (e *SyntaxErrorWithSuggestedFix) EndPosition(_ common.MemoryGauge) ast.Position {
+	return e.Pos
+}
+
+func (e *SyntaxErrorWithSuggestedFix) Error() string {
+	return e.Message
+}
+
+func (e *SyntaxErrorWithSuggestedFix) SuggestFixes(_ string) []errors.SuggestedFix[ast.TextEdit] {
+	return []errors.SuggestedFix[ast.TextEdit]{
+		{
+			Message: fmt.Sprintf("replace with %s", e.SuggestedFix),
+			TextEdits: []ast.TextEdit{
+				{
+					Replacement: e.SuggestedFix,
+					Range: ast.Range{
+						StartPos: e.Pos,
+						EndPos:   e.Pos,
+					},
```

Comment:
Not sure, but isn't this currently suggesting to replace "nothing" (start and and position are the same),so is effectively suggesting an insert? Maybe replace the single position (`Pos`) with a range.